src/chatClient.cpp

```diff
@@ -3663,4 +3731,352 @@ std::string encodeFirstName(const std::string& first)
     return result;
 }
 
+// Init Stats metods
+
+void InitStats::setNumNodes(long long numNodes)
+{
+    mNumNodes = numNodes;
+}
+
+void InitStats::setNumContacts(long numContacts)
+{
+    mNumContacts = numContacts;
+}
+
+void InitStats::setNumChats(long numChats)
+{
+    mNumChats = numChats;
+}
+
+bool InitStats::isFinished() const
+{
+    return mFinished;
+}
+
+void InitStats::setFinished(bool finished)
+{
+    mFinished = finished;
+    clearStats();
+}
+
+InitStats::InitStats()
+{
+    mNumNodes = 0;
+    mNumChats = 0;
+    mNumContacts = 0;
+    mTotalElapsed = 0;
+    mFinished = false;
+    mInitState = kInitNewSession;
+    KR_LOG_INFO("Init stats created");
+}
+
+void InitStats::clearStats()
```

Comment:
Maybe it's better to declare a `init()`/`reset()` method and call it from ctor and from `onCompleted()`